Strong start categories

3ddiehead opened the pull request:
Fully functional with automated test document

jbellik commented on constraints/strongStart.js:
```diff
@@ -214,3 +291,32 @@ function strongStartInit(stree, ptree, cat){
+	}
+}
+
+//Recurses down the first children a node to check whether or not it dominates a node with the specified id at its left edge.
```
Improve this comment as well. "A helper function for [function name]. Takes [argument list and info]. Returns true if [whatever conditions]. Returns false otherwise."

Please change the return values to true/false instead of using 1/0.

jbellik replied:
I think this function name could be more transparent as well. hasIdAtLeftEdge(node, id)?

jbellik commented on test/strongStartHsuTest.html:
```diff
@@ -0,0 +1,506 @@
+    <div id="mocha">
+        StrongStart(k/p) Testing
+    </div>
+    <p>StrongStart(k/p) tableaux begin on page 95</p><p>https://ling.auf.net/lingbuzz/003512</p>
```
Make this a link.

jbellik commented on test/strongStartHsuTest.html:
```diff
@@ -0,0 +1,506 @@
+
+        describe('Tableau (59) in Hsu 2016', function() {
+
+            it('Hsu_Tree5', function() {
```
For each it() statement, please include parenthesizeTree(whateverTheRelevantPtreeIs) in the description part.
e.g.: it('Hsu_Tree5: '+parenthesizeTree(ptree6), function(){...}) in this case.

jbellik reviewed (changes requested):
This looks like it's working well. The test file is great. I'd just like a few changes for readability, mostly in comments but also potentially a couple of function names. Good work!

3ddiehead commented:
It appears that putting in two arguments for strongStart_Hsu is going to be a problem with the JSON string to function-call. For example, when calling the function for the creation of a tableau, 'strongStart_Hsu-w-i' does not work.  The system indicates that the "i" was unexpected.  I could change the function to fit within the system but it might be a little less clean.  Maybe making the categories into options would work?  What do you all think?

jbellik commented:
Ah yes, I forgot about this issue. I was thinking we could make a couple of
wrapper functions. In fact, I decided it would be easier to just write them
than to try to explain them; see the commit I just pushed to this branch.
This is the simplest solution, although there are definitely other ways we
could handle this (with an array of categories, for instance). Try those in
a tableau and see if they work ok.

On Wed, Jan 20, 2021 at 2:37 PM 3ddiehead <notifications@github.com> wrote:

> It appears that putting in two arguments for strongStart_Hsu is going to
> be a problem with the JSON string to function-call. For example, when
> calling the function for the creation of a tableau, 'strongStart_Hsu-w-i'
> does not work.  The system indicates that the "i" was unexpected. I could
> change the function to fit within the system but it might be a little less
> clean.  Maybe making the categories into options would work? What do you
> all think?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/syntax-prosody-ot/main/pull/546#issuecomment-763998379>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC7ROKW4JOXA4O3WISJHSWDS25LK3ANCNFSM4WH4NNKQ>
> .
>


-- 


--
Jennifer Bellik, PhD
Post-doctoral researcher & lecturer
UC Santa Cruz
https://people.ucsc.edu/~jbellik/

3ddiehead commented:
Those changes work perfectly, here is what I propose for the categories available to k and p:
k = syllable, foot, word, phi
p = phi, iota (maybe u, but what is it?)
Let me know if this sounds good.

jbellik commented:
We don't actually need 'u'. It was a mistake to include it in the prosodic
hierarchy since Junko & Armin have proposed a PH that doesn't have
"utterance" in it at all. So for the interface, that selection of values
for k looks good, and then we can have distinct entries for p = phi, p = i.
These should all go under the "show more" section in Sisterhood.

On Wed, Jan 20, 2021 at 5:10 PM 3ddiehead <notifications@github.com> wrote:

> Those changes work perfectly, here is what I propose for the categories
> available to k and p:
> k = syllable, foot, word, phi
> p = phi, iota (maybe u, but what is it?)
> Let me know if this sounds good.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/syntax-prosody-ot/main/pull/546#issuecomment-764144751>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC7ROKUE5TCIM4XJ5DRQ4ADS255G5ANCNFSM4WH4NNKQ>
> .
>


-- 


--
Jennifer Bellik, PhD
Post-doctoral researcher & lecturer
UC Santa Cruz
https://people.ucsc.edu/~jbellik/

3ddiehead commented:
Alright, added to the interface. Also, there was a minor error in the code for StrongStart (any sister) where the value of a checkbox was "syll" while text read as &iota;. I changed the readout to &sigma; instead because I assume that's what it was supposed to be.

jbellik commented:
Oh, thanks for correcting that. Whoops.

On Thu, Jan 21, 2021 at 1:30 PM 3ddiehead <notifications@github.com> wrote:

> Alright, added to the interface. Also, there was a minor error in the code
> for StrongStart (any sister) where the value of a checkbox was "syll" while
> text read as ι. I changed the readout to σ instead because I assume that's
> what it was supposed to be.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/syntax-prosody-ot/main/pull/546#issuecomment-764953331>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC7ROKSQUWC64GP4HDN6JZTS3CMFXANCNFSM4WH4NNKQ>
> .
>


-- 


--
Jennifer Bellik, PhD
Post-doctoral researcher & lecturer
UC Santa Cruz
https://people.ucsc.edu/~jbellik/

jbellik commented on constraints/strongStart.js:
```diff
@@ -1,6 +1,36 @@
+function seekCatLeftEdge(node, cat, id){
```
I think this and searchDownLeft have been superceded by catDomsIdAtLeftEdge and its helper, hasIdAtLeftEdge, no? Looks like you forgot to delete these, or some kind of versioning weirdness happened.

3ddiehead replied:
Haha, not sure how that happened

jbellik commented on constraints/strongStart.js:
```diff
@@ -36,6 +68,62 @@ function strongStart_Elfner(s, ptree, k){
+	
+	// if node.children[0].cat === k and has a sibling, then compare it with its sibling as well as for domination by a node of cat p along the left edge.
+	if(node.children.length>1 && node.children[0].cat === k){		
+		if((pCat.isLower(node.children[0].cat, node.children[1].cat)) && seekCatLeftEdge(ptree, p, node.id)){ // searches tree for node of cat p dominating this node of cat k
```
Shouldn't this be calling catDomsIdAtLeftEdge instead?

3ddiehead replied:
Yes, you're right
